src/modules/exec.ts

```diff
@@ -61,15 +66,16 @@ export async function awaitSpawn(
         };
         process.on('SIGINT', sigintCallback);
         ptyProcess.onExit((code) => {
-            if (process.stdin) {
-                process.stdin.unref();
+            if (interactive) {
+                process.stdin.setRawMode(false);
             }
+            process.stdin.pause();
```

Comment:
Why is pause needed?

Reply:
it achieves basically the same as unref() but seems to be more appropriate because it stops reading from stdin after the process has exited, ensuring no more input is sent to a non-existent process. unref() doesn't stop the data flow it just indicates to the Node.js event loop that this stream should not prevent the process from exiting.